File: -- src_R/| |-- 03_akar_kuadrat.R
#Program 3
#Fungsi Akar-akar Persamaan Kuadrat
hitung_akar <- function(a, b, c) {
  diskriminan <- b^2 - 4*a*c
  if (diskriminan > 0) {
    x1 <- (-b + sqrt(diskriminan)) / (2*a)
    x2 <- (-b - sqrt(diskriminan)) / (2*a)
    cat("Persamaan Memiliki Dua Akar Real yang Berbeda:\n")
    cat("x1 =", sprintf("%.3f", x1), "\n")
    cat("x2 =", sprintf("%.3f", x2), "\n")
  } else if (diskriminan == 0) {
    x <- -b / (2*a)
    cat("Persamaan Memiliki Akar Real Kembar:\n")
    cat("x1 = x2 =", sprintf("%.3f", x), "\n")
  } else {
    real <- -b / (2*a)
    imaj <- sqrt(abs(diskriminan)) / (2*a)
    cat("Persamaan Memiliki Akar-Akar Imajiner:\n")
    cat("x1 =", sprintf("%.3f", real), "+", sprintf("%.3f", imaj), "i\n")
    cat("x2 =", sprintf("%.3f", real), "-", sprintf("%.3f", imaj), "i\n")
  }
}

#Memasukkan Nilai Variabel dari Persamaan Kuadrat
a <- as.numeric(readline("Masukkan nilai a: "))
b <- as.numeric(readline("Masukkan nilai b: "))
c <- as.numeric(readline("Masukkan nilai c: "))

#Menampilkan Bentuk Persamaan
cat("Persamaan:", a, "x² +", b, "x +", c, "= 0\n")

#Menampilkan Hasil 
hitung_akar(a, b, c)

File: -- src_python/| |-- 03_akar_kuadrat.py
#Program 3
#Fungsi Akar-Akar Persamaan Kuadrat
def hitung_akar(a, b, c):
    d = b**2 - 4*a*c
    if d > 0:
        print("Persamaan Memiliki Dua Akar Real yang Berbeda:")
        print("x1 =", round((-b + math.sqrt(d)) / (2*a),3))
        print("x2 =", round((-b - math.sqrt(d)) / (2*a),3))
    elif d == 0:
        x = -b / (2*a)
        print("Persamaan Memiliki Akar Real Kembar:")
        print("x1 = x2 =", round(x),3)
    else:
        real = -b / (2*a)
        imaj = math.sqrt(abs(d)) / (2*a)
        print("Persamaan Akar-Akar Imajiner:")
        print("x1 =", round(real, 3), "+", round(imaj, 3), "i")
        print("x2 =", round(real, 3), "-", round(imaj, 3), "i")

#Memasukkan Nilai Variabel dari Persamaan Kuadrat
a = float(input("Masukkan nilai a: "))
b = float(input("Masukkan nilai b: "))
c = float(input("Masukkan nilai c: "))

#Menampilkan Bentuk Persamaan
print("Persamaan:", a, "x² +", b, "x +", c, "= 0")

#Menampilkan Hasil
hitung_akar(a, b, c)
